### PR TITLE
Support Notifications for Android 12 and lower for both data and notifications messages (console)

### PR DESCRIPTION
**Pull request by cw-sanjeev**

### CODE REVIEW

**Review by cw-suryansh — APPROVED** [review]

File: lib/android/app/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule.java [hunk 1]
```diff
@@ -89,17 +90,14 @@ public void getInitialNotification(final Promise promise) {
             }
 
             result = Arguments.fromBundle(notification.asBundle());
```
Comment (CW-RanjitMahadik) [inline lib/android/app/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule.java]: why did we remove it?

File: lib/android/app/src/main/java/com/wix/reactnativenotifications/core/NotificationIntentAdapter.java [hunk 2]
```diff
@@ -1,34 +1,54 @@
...
-            return taskStackBuilder.getPendingIntent((int) System.currentTimeMillis(), PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_ONE_SHOT | PendingIntent.FLAG_IMMUTABLE);
+            intent = appLaunchHelper.getLaunchIntent(appContext);
+            intent.putExtra(PUSH_NOTIFICATION_EXTRA_NAME, notification.asBundle());
+            intent.putExtra(Defs.IS_INTENT_HANDLED, false);
         }
+        return intent;
+    }
+
+    public static boolean cannotHandleTrampolineActivity(Context appContext) {
+        return android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.R && appContext.getApplicationInfo().targetSdkVersion >= 31;
```
Comment (CW-RanjitMahadik) [inline lib/android/app/src/main/java/com/wix/reactnativenotifications/core/NotificationIntentAdapter.java:47]:
```suggestion
        return cannotHandleTrampolineActivity() && appContext.getApplicationInfo().targetSdkVersion >= 31;
```

File: lib/android/app/src/main/java/com/wix/reactnativenotifications/core/NotificationIntentAdapter.java [hunk 3]
```diff
@@ -1,34 +1,54 @@
...
-            Intent mainActivityIntent = appContext.getPackageManager().getLaunchIntentForPackage(appContext.getPackageName());
-            mainActivityIntent.putExtra(PUSH_NOTIFICATION_EXTRA_NAME, notification.asBundle());
-            TaskStackBuilder taskStackBuilder = TaskStackBuilder.create(appContext);
-            taskStackBuilder.addNextIntentWithParentStack(mainActivityIntent);
-            return taskStackBuilder.getPendingIntent((int) System.currentTimeMillis(), PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_ONE_SHOT | PendingIntent.FLAG_IMMUTABLE);
+            intent = appLaunchHelper.getLaunchIntent(appContext);
+            intent.putExtra(PUSH_NOTIFICATION_EXTRA_NAME, notification.asBundle());
+            intent.putExtra(Defs.IS_INTENT_HANDLED, false);
         }
+        return intent;
```
Comment (CW-RanjitMahadik) [inline lib/android/app/src/main/java/com/wix/reactnativenotifications/core/NotificationIntentAdapter.java:43]:
```suggestion
        intent.putExtra(Defs.IS_INTENT_HANDLED, false);
        return intent;
```

File: lib/android/app/src/main/java/com/wix/reactnativenotifications/core/NotificationIntentAdapter.java [hunk 4]
```diff
@@ -1,34 +1,54 @@
...
+import android.os.Build;
 import android.os.Bundle;
 
+import com.wix.reactnativenotifications.Defs;
 import com.wix.reactnativenotifications.core.notification.PushNotificationProps;
 
 public class NotificationIntentAdapter {
     private static final String PUSH_NOTIFICATION_EXTRA_NAME = "pushNotification";
 
+    private Application application;
```
Comment (CW-RanjitMahadik) [inline lib/android/app/src/main/java/com/wix/reactnativenotifications/core/NotificationIntentAdapter.java:17]: are we using application anywhere in this file?

File: lib/android/app/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule.java (hunk 1 above)
Comment (cw-sanjeev) [reply lib/android/app/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule.java]: App navigates to notification payload again taking it from intent, this has been tested

File: lib/android/app/src/main/java/com/wix/reactnativenotifications/core/NotificationIntentAdapter.java (hunk 2 above)
Comment (cw-sanjeev) [reply lib/android/app/src/main/java/com/wix/reactnativenotifications/core/NotificationIntentAdapter.java:47]: done

File: lib/android/app/src/main/java/com/wix/reactnativenotifications/core/NotificationIntentAdapter.java (hunk 3 above)
Comment (cw-sanjeev) [reply lib/android/app/src/main/java/com/wix/reactnativenotifications/core/NotificationIntentAdapter.java:43]: done

File: lib/android/app/src/main/java/com/wix/reactnativenotifications/core/NotificationIntentAdapter.java (hunk 4 above)
Comment (cw-sanjeev) [reply lib/android/app/src/main/java/com/wix/reactnativenotifications/core/NotificationIntentAdapter.java:17]: done

File: lib/android/app/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule.java [hunk 5]
```diff
@@ -63,23 +63,24 @@ public void onActivityResult(Activity activity, int requestCode, int resultCode,
...
+                intent.putExtra(Defs.IS_INTENT_HANDLED, true);
                 notification.onOpened();
             }
         }
     }
 
     @ReactMethod
     public void refreshToken() {
-        if(BuildConfig.DEBUG) Log.d(LOGTAG, "Native method invocation: refreshToken()");
+        if (BuildConfig.DEBUG) Log.d(LOGTAG, "Native method invocation: refreshToken()");
```
Comment (cw-atharva) [inline lib/android/app/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule.java:77]: why are we logging this?

File: lib/android/app/src/main/java/com/wix/reactnativenotifications/core/NotificationIntentAdapter.java (hunk 2 above)
Comment (cw-atharva) [inline lib/android/app/src/main/java/com/wix/reactnativenotifications/core/NotificationIntentAdapter.java]: can we make 31 constant ? does we do this in java?

**Review by CW-RanjitMahadik — APPROVED** [review]

**Pull request merged**